op-dispute-mon/mon/service.go

```diff
@@ -144,7 +144,13 @@ func (s *Service) initExtractor(cfg *config.Config) {
 		extract.NewBalanceEnricher(),
 		extract.NewL1HeadBlockNumEnricher(s.l1Client),
 		extract.NewSuperAgreementEnricher(s.logger, s.metrics, s.supervisorClient, clock.SystemClock),
-		extract.NewOutputAgreementEnricher(s.logger, s.metrics, s.rollupClient, clock.SystemClock),
+		extract.NewOutputAgreementEnricher(s.logger, s.metrics, func() []extract.OutputRollupClient {
+			clients := make([]extract.OutputRollupClient, len(s.rollupClients))
+			for i, client := range s.rollupClients {
+				clients[i] = client
+			}
+			return clients
+		}(), clock.SystemClock),
```

Comment:
The embedded function is clever, but probably clearer to just extract it as a named private function like `func asOutputRollupClient`.